nucleus/cluster/admin/src/main/java/com/sun/enterprise/v3/admin/cluster/StartInstanceCommand.java

```diff
@@ -277,21 +294,29 @@ private void startInstance(AdminCommandContext ctx) {
     }
 
     // return null means A-OK
-    private String pollForLife(Server instance) {
-        int counter = 0;  // 120 seconds
-
-        while (++counter < 240) {
-            if (instance.isRunning())
-                return null;
-
-            try {
-                Thread.sleep(500);
-            }
-            catch (Exception e) {
-                // ignore
+    private String pollForLife(final Server instance) {
+
+        // Start a new thread to check when the instance has started
+        final CountDownLatch instanceTimeout = new CountDownLatch(1);
+        ScheduledExecutorService executor = Executors.newScheduledThreadPool(1);
+        executor.schedule(new Runnable() {
+            @Override
+            public void run() {
+                if (instance.isRunning()) {
+                    instanceTimeout.countDown();
+                }
             }
+        }, 500, TimeUnit.MILLISECONDS);
+
+        // If the timeout is reached, return the timeout message. Otherwise return null (success).
+        try {
+			instanceTimeout.await(timeout, TimeUnit.SECONDS);
+		} catch (InterruptedException e) {
+			return Strings.get("start.instance.timeout", instanceName);
+		} finally {
+            executor.shutdown();
```

Comment:
Maybe put the "try" higher up? If `executor.schedule` throws an exception, the executor might not be shut down.

Reply:
Good point, I've moved it higher up